### DISCUSSION TITLE
Inverse filter, delete if it doesn't match the filter

### DISCUSSION THREAD

**Discussion by Liggliluff** [victornpb/undiscord#447] [3. Ideas]
The current filters only allows you to delete messages that has containing text, has link, has file, and/or is pinned.

I wish to have the option to keep messages that contains a certain text, has link, has file and/or is pinned.

One idea for the UI design could be:

```
▼ Filter
SEARCH
[Containing text]
⦿ Delete if it contains the text
ⵔ Keep if it contains the text

Delete messages that:
LINK     ( either ) (  with  ) (without)
FILE     ( either ) (  with  ) (without)
PINNED   ( either ) (  with  ) (without)
MENTION  ( either ) (  with  ) (without)
REACTION ( either ) (  with  ) (without)
```
either = regardless if it's with or without, delete it. with = only if it has. without = only if it does not have.

**Comment by last1melody** [comment]
I've added a modified version of Tampermonkey that adds mention as a new filter, hope this helps!
[undiscord_with_mention_quotes.zip](https://github.com/victornpb/undiscord/files/14622567/undiscord_with_mention_quotes.zip)
![image](https://github.com/victornpb/undiscord/assets/116696935/b032ae65-ebd8-4699-88a6-1de11640c1cb)
